.github/workflows/pr.yml

```diff
@@ -44,36 +44,59 @@ jobs:
     uses: ./.github/workflows/dispatch-build-and-test.yml
     strategy:
       fail-fast: false
-      matrix: 
-        cuda_version: ${{ fromJSON(needs.compute-matrix.outputs.CUDA_VERSIONS) }} 
+      matrix:
+        cuda_version: ${{ fromJSON(needs.compute-matrix.outputs.CUDA_VERSIONS) }}
         compiler: ${{ fromJSON(needs.compute-matrix.outputs.COMPILERS) }}
     with:
       per_cuda_compiler_matrix: ${{ toJSON(fromJSON(needs.compute-matrix.outputs.PER_CUDA_COMPILER_MATRIX)[ format('{0}-{1}', matrix.cuda_version, matrix.compiler) ]) }}
       build_script: "./ci/build_cub.sh"
       test_script: "./ci/test_cub.sh"
-  
+
   libcudacxx:
     name: libcudacxx CUDA${{ matrix.cuda_version }} ${{ matrix.compiler }}
     needs: compute-matrix
     uses: ./.github/workflows/dispatch-build-and-test.yml
     strategy:
       fail-fast: false
-      matrix: 
-        cuda_version: ${{ fromJSON(needs.compute-matrix.outputs.CUDA_VERSIONS) }} 
+      matrix:
+        cuda_version: ${{ fromJSON(needs.compute-matrix.outputs.CUDA_VERSIONS) }}
         compiler: ${{ fromJSON(needs.compute-matrix.outputs.COMPILERS) }}
     with:
       per_cuda_compiler_matrix: ${{ toJSON(fromJSON(needs.compute-matrix.outputs.PER_CUDA_COMPILER_MATRIX)[ format('{0}-{1}', matrix.cuda_version, matrix.compiler) ]) }}
       build_script: "./ci/build_libcudacxx.sh"
-      test_script: "./ci/test_libcudacxx.sh" 
+      test_script: "./ci/test_libcudacxx.sh"
+
+  nvrtc:
+    name: libcudacxx nvrtc
+    needs: compute-matrix
+    uses: ./.github/workflows/dispatch-build-and-test.yml
+    strategy:
+      fail-fast: false
+      matrix:
+        cuda_version: ${{ fromJSON(needs.compute-matrix.outputs.CUDA_VERSIONS) }}
+        compiler: ${{ fromJSON(needs.compute-matrix.outputs.COMPILERS) }}
+        exclude:
+          - cuda_version: "11.1"
+          - cuda_version: "12.1"
+          - compiler: clang
+          - compiler: gcc
+        include:
+          - cuda_version: "12.1"
+          - compiler: gcc-12
+    with:
+      per_cuda_compiler_matrix: ${{ toJSON(fromJSON(needs.compute-matrix.outputs.PER_CUDA_COMPILER_MATRIX)[ format('{0}-{1}', matrix.cuda_version, matrix.compiler) ]) }}
+      build_script: "./ci/build_libcudacxx_nvrtc.sh"
+      test_script: "./ci/test_libcudacxx_nvrtc.sh"
```

Comment:
I'm thinking that we don't really need separate build/test jobs for the NVRTC testing.

The only reason I did that for other environments is to preseed sccache with the build artifacts using CPU-only build machines to avoid paying the full compile cost on GPU machines (which are a scarcer resource). 

Given NVRTC doesn't work with sccache, there's no need to do that here.